keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -134,256 +137,258 @@ When a workload (app pod, deployment, configs) is moved to another cluster, as l
-1. `protocol.name`: (required) specifies the desired protocol.  One of {“s3”, “gs”, or “azureBlob”}.
-1. `protocol.version`: (optional) specifies the desired version of the `protocol`. For "s3", a value of "v2" or "v4" could be used. 
-1. `anonymousAccessMode`: (optional) a string specifying *uncredentialed* access to the backend bucket.  This is applicable for cases where the backend storage is intended to be publicly readable and/or writable. One of:
-   - "private": Default, disallow uncredentialed access to the backend storage.
-   - "publicReadOnly": Read only, uncredentialed users can call ListBucket and GetObject.
-   - "publicWriteOnly": Write only, uncredentialed users can only call PutObject.
-   - "publicReadWrite": Read/Write, same as `ro` with the addition of PutObject being allowed.
-1. `retentionPolicy`: defines bucket retention for greenfield `BucketRequest`.
-   - _Retain_: (default) the `Bucket` instance and the backend bucket are preserved. The `Bucket` may potentially be reused.
-   - _Delete_: the backend bucket and the `Bucket` instance are deleted.
+1. `isDefault`: boolean annotaion. If omitted the setting is false. If present but lacking a value the setting is true. If true then a `BucketRequest` may omit the `BucketClass` reference.
+1. `provisioner`: (optional) the name of the vendor-specific driver supporting the `protocol`. If empty then there is no driver/sidecar, and the admin has to create the `Bucket` and `BucketAccess` instances manually.
+1. `protocol`: protocol-specific field the application will use to access the backend storage. The admin is expected to only include the desired protocol name (e.g. "s3") and not the other details listed. Note: these protocol details are defined in the `Bucket` instance.
+     - `azureBlob`: data required to target a provisioned azure container.
+     - `gs`: data required to target a provisioned GS bucket.
+     - `s3`: data required to target a provisioned S3 bucket.
+1. `deletionPolicy`: (required) defines the outcome when a greenfield `BucketRequest` is deleted, and is ignored for brownfield buckets.
+   - _Retain_: keep both the `Bucket` instance and the backend bucket and its contents. The `Bucket` instance may be marked unavailable and it may potentially be reused (post MVP).
+   - _Delete_: when all access to the `Bucket` is complete, delete both the `Bucket` instance and the backend bucket, including all content in that bucket. If, for example, a running pod or a `BucketAccess` instance refer to the `Bucket` instance, then the delete is deferred until all accessors have been deleted.
+   - _ForceDelete_: delete both the `Bucket` instance and the backend bucket, including all content in that bucket, ignoring any accessors or workloads that may be referencing the `Bucket`.
```

Comment:
Same question here - what happens to BucketAccess referring to the deleted Bucket